hub/core/dataset/dataset.py

```diff
@@ -1343,6 +1343,70 @@ def _set_read_only(self, value: bool, err: bool):
     def read_only(self, value: bool):
         self._set_read_only(value, True)
 
+    def cleanlab(
+        self,
+        module = None,
+        criterion = None,
+        optimizer = None,
+        optimizer_lr: int = 0.01,
+        device: str = "cpu",
```

Comment:
I'm thinking of making the `device` an optional parameter and doing a check like `device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")` in `pytorch_module_to_skorch()`. Also can add `"mps" if torch.backends. mps. is_available()` since they've recently added M1  support. Would this approach be sound?